internal/security/secretstore/init.go

```diff
@@ -59,132 +62,159 @@ func NewBootstrap(insecureSkipVerify bool, vaultInterval int) *Bootstrap {
+					return true
+				}
+
+				if secretStoreConfig.RevokeRootTokens {
 					// Never persist the root token to disk on secret store initialization if we intend to revoke it later
 					initResponse.RootToken = ""
 					lc.Info("Root token stripped from init response for security reasons")
 				}
-				_, err = vc.Unseal(&initResponse)
+
+				err = client.Unseal(initResponse.Keys, initResponse.KeysBase64)
 				if err == nil {
-					shouldContinue = false
+					lc.Errorf("Unable to unseal Vault: %s", err.Error())
+					return false
 				}
+
 				// We need the unencrypted initResponse in order to generate a temporary root token later
 				// Make a copy and save the copy, possibly encrypted
-				encryptedInitResponse := initResponse
+				var encryptedInitResponse types.InitResponse
 				// Optionally encrypt the vault init response based on whether encryption was enabled
 				if vmkEncryption.IsEncrypting() {
-					if err := vmkEncryption.EncryptInitResponse(&encryptedInitResponse); err != nil {
-						lc.Error(fmt.Sprintf("failed to encrypt init response from secret store: %s", err.Error()))
+					encryptedInitResponse, err = vmkEncryption.EncryptInitResponse(initResponse)
+					if err != nil {
+						lc.Errorf("failed to encrypt init response from secret store: %s", err.Error())
 						return false
 					}
 				}
-				if err := saveInitResponse(lc, fileOpener, configuration.SecretService, &encryptedInitResponse); err != nil {
-					lc.Error(fmt.Sprintf("unable to save init response: %s", err.Error()))
+				if err := saveInitResponse(lc, fileOpener, secretStoreConfig, &encryptedInitResponse); err != nil {
+					lc.Errorf("unable to save init response: %s", err.Error())
 					return false
 				}
+
 			case http.StatusServiceUnavailable:
-				lc.Info(fmt.Sprintf("vault is sealed (status code: %d). Starting unseal phase", sCode))
-				if err := loadInitResponse(lc, fileOpener, configuration.SecretService, &initResponse); err != nil {
-					lc.Error(fmt.Sprintf("unable to load init response: %s", err.Error()))
+				lc.Infof("vault is sealed (status code: %d). Starting unseal phase", sCode)
+				if err := loadInitResponse(lc, fileOpener, secretStoreConfig, &initResponse); err != nil {
+					lc.Errorf("unable to load init response: %s", err.Error())
 					return false
 				}
 				// Optionally decrypt the vault init response based on whether encryption was enabled
 				if vmkEncryption.IsEncrypting() {
-					if err := vmkEncryption.DecryptInitResponse(&initResponse); err != nil {
-						lc.Error(fmt.Sprintf("failed to decrypt key shares for sercret store unsealing: %s", err.Error()))
+					initResponse, err = vmkEncryption.DecryptInitResponse(initResponse)
+					if err != nil {
+						lc.Errorf("failed to decrypt key shares for secret store unsealing: %s", err.Error())
 						return false
 					}
 				}
-				_, err := vc.Unseal(&initResponse)
+				err := client.Unseal(initResponse.Keys, initResponse.KeysBase64)
```

Comment:
@lenny-intel if you're not going to pass in the whole thing then please choose one -- either Keys or KeysBase64

Reply:
K, will adjust Unseal to just take the KeysBase64, since that is what it is inly using. ;-)

Reply:
Done